public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -0,0 +1,585 @@
+                    isSelected={groupSelected(option.value) || groupPartiallySelected(option.value)}
+                    partiallySelected={groupPartiallySelected(option.value)}
+                    onChange={onGroupChange}
+                    onClick={onMenuGroupClick}
+                  />
+                ))}
+              </div>
+            </div>
+          ) : (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {fixedRoles.map((option, i) => (
+                  <RoleMenuOption
+                    data={option}
+                    key={i}
+                    isSelected={
+                      !!(
+                        (option.uid && !!selectedOptions.find((opt) => opt.uid === option.uid)) ||
+                        !!selectedBuiltInRoles.find((role) => role.uid === option.uid)
+                      )
+                    }
+                    disabled={!!(option.uid && !!selectedBuiltInRoles.find((role) => role.uid === option.uid))}
+                    onChange={onChange}
+                    // onClick={onMenuGroupClick}
+                    hideDescription
+                  />
+                ))}
+              </div>
+            </div>
+          )}
+          {!!customRoles.length && (
+            <div>
+              <div className={customStyles.groupHeader}>Custom roles</div>
+              <div className={styles.optionBody}>
+                {customRoles.map((option, i) => (
+                  <RoleMenuOption
+                    data={option}
+                    key={i}
+                    isSelected={
+                      !!(
+                        (option.uid && !!selectedOptions.find((opt) => opt.uid === option.uid)) ||
+                        !!selectedBuiltInRoles.find((role) => role.uid === option.uid)
+                      )
+                    }
+                    disabled={!!(option.uid && !!selectedBuiltInRoles.find((role) => role.uid === option.uid))}
+                    onChange={onChange}
+                    hideDescription
+                  />
+                ))}
+              </div>
+            </div>
+          )}
+        </CustomScrollbar>
+        <div className={customStyles.menuButtonRow}>
+          <HorizontalGroup justify="flex-end">
+            <Button size="sm" fill="text" onClick={onClearInternal}>
+              Clear all
+            </Button>
+            <Button size="sm" onClick={onUpdateInternal}>
+              Update
+            </Button>
+          </HorizontalGroup>
+        </div>
+      </div>
+      {showSubMenu ? (
+        <RolePickerSubMenu
+          options={subMenuOptions}
+          selectedOptions={selectedOptions}
+          disabledOptions={selectedBuiltInRoles}
+          onSelect={onChange}
+          onClear={onClearSubMenu}
+        />
+      ) : (
+        <div></div>
+      )}
```

Comment:
```suggestion
      ) : null}
```
Not sure there's a need for an empty div.